app/models/manageiq/providers/vmware/infra_manager/refresh_parser_inventory_object.rb

```diff
@@ -82,7 +82,9 @@ def storage_inv_to_hashes(inv)
           )
         end
 
-        result << result.new_inventory_object(new_result)
+        new_result = result.new_inventory_object(new_result)
```

Comment:
result << result.new_inventory_object(new_result)

should work, was there an issue with it?